numba/_typeof.c

```diff
@@ -831,6 +834,52 @@ int typecode_arrayscalar(PyObject *dispatcher, PyObject* aryscalar) {
     return BASIC_TYPECODES[typecode];
 }
 
+static
+int typecode_devicendarray(PyObject *dispatcher, PyObject *ary)
+{
+    int typecode;
+    int dtype;
+    int ndim;
+    int layout = 0;
+
+    PyObject* flags = PyObject_GetAttrString(ary, "flags");
```

Comment:
unchecked call, also leaks if there's an issue later?

Reply:
Added a check and a corresponding decref.

Reply:
thanks